backend/src/schema/resolvers/reports.js

```diff
@@ -11,12 +11,32 @@ export default {
         description: description,
       }
 
+      const reportQueryRes = await session.run(
+        `
+        match (u:User {id:$submitterId}) -[:REPORTED]->(report)-[:REPORTED]-> (resource {id: $resourceId}) 
+        return  report, labels(resource)[0] as type
+        `,
+        {
+          resourceId: id,
+          submitterId: user.id,
+        },
+      )
+      const [rep] = reportQueryRes.records.map(record => {
+        return {
+          report: record.get('report'),
+          type: record.get('type'),
+        }
+      })
+
+      if (rep) {
+        throw new Error(rep.type)
```

Comment:
the error is output if a report already exists. 
throw is as I understood it equal to return. 
rep.type is the type of the report (user,post or comment). 
new Error only outputs the string . no additional information that would otherwise come when an error is reported from the database. 
in the frontend the respective error message is displayed. 
throw aborts the code like return.